### ISSUE TITLE
How to add Theme app extension using laravel osiset Package

### ISSUE THREAD

**Issue by yasir-naseer** [osiset/laravel-shopify#1004]
Greetings, due to recent changes in shopify, the shopify app review team ask the developers to add theme app extension instead to script tag, so i am wondering how this can be achieved in this package, as most of the online resources do it using Shopify CLI which mostly work with node.js

Any help would be highly appreciated. thanks

**Comment by brucelee90** [comment]
It seems like this [https://github.com/osiset/laravel-shopify/issues/980](https://github.com/osiset/laravel-shopify/issues/980) thread didn't help you..

Can you elaborate where you get stuck? What have you tried so far and what is not working for you?

**Comment by yasir-naseer** [comment]
Yes, actually i dont understand the process of adding app blocks to support shopify 2.0 theme app extension requirements from shopify app approval team.

When i searched about app block, most of the tutorials talks about shopify CLI which works with node.js, so i am not sure how to achieve app block feature in this package or simply how to add app block using this package

Previously before shopify 2.0, all i do is use scripttag, to add code sections (HTML, JS) at run time using ajax. but seems now shopify wants us to display the added widget in theme customizer and so on. 

So basically if i can have some help on how to add app block, add theme customizer preview of added blocks this will help me get my app approved (Note: All my app do is it shows a HTML banner at cart page which comes from app via script tag)

**Comment by brucelee90** [comment]
So creating the app block is actually straight forward. You can use this link right here: [https://shopify.dev/apps/online-store/theme-app-extensions/getting-started](https://shopify.dev/apps/online-store/theme-app-extensions/getting-started
). you just create a directory for the theme app extension in your project and push it to your app on shopify.

As you said, now they want us to display a widget in the customizer which would then make the usage of scripttags obsolete. You could also try submit your without using any scripttags at all, but then the app would only be usable for users of  shopify theme 2.0.

What confuses me as well is the whole part with the ajax request, because we still need the ajax requests. If we change some settings within the widgets, then all of the data is saved in the meta fields. This data is then read by the liquid file. But we can not store that data in a database anymore. But if that is the case, then why should we make GUI for our apps anymore? :D That's really confusing tbh.

**Comment by damquanghieu** [comment]
> So creating the app block is actually straight forward. You can use this link right here: https://shopify.dev/apps/online-store/theme-app-extensions/getting-started. you just create a directory for the theme app extension in your project and push it to your app on shopify.
> 
> As you said, now they want us to display a widget in the customizer which would then make the usage of scripttags obsolete. You could also try submit your without using any scripttags at all, but then the app would only be usable for users of shopify theme 2.0.
> 
> What confuses me as well is the whole part with the ajax request, because we still need the ajax requests. If we change some settings within the widgets, then all of the data is saved in the meta fields. This data is then read by the liquid file. But we can not store that data in a database anymore. But if that is the case, then why should we make GUI for our apps anymore? :D That's really confusing tbh.

I'm having a lot of questions with 2.0 extensions.
If I write all my code in the extension, what else does the app need?
Or use your application to communicate with the extension?
Who knows about this please help me!

**Comment by thebleshbanz** [comment]
Your app is currently using ScriptTag API. Update your app to theme app
extensions to ensure compatibility with Online Store 2.0 themes.

App Review Team also made an issue of scriptTag on my app.

please share solution.


Thanks a lots

**Comment by apurbajnu** [comment]
You may follow this link. This is working for me. Hopefully, it will work for you. 
https://github.com/osiset/laravel-shopify/issues/980#issuecomment-991649153

**Comment by tanseercena** [comment]
@yasir-naseer  I think this will help you: https://github.com/osiset/laravel-shopify/issues/1167#issuecomment-1202443351

**Comment by Kyon147** [comment]
Closing this as decent solution provided.
